packages/api/portfolios/environments/updateEnvironment.ts

```diff
@@ -0,0 +1,78 @@
+import "reflect-metadata";
+import { createConnection } from "../../utils/database";
+import { Portfolio } from "../../../orm/entity/Portfolio";
+import { Application } from "../../../orm/entity/Application";
+import { Environment, IEnvironment } from "../../../orm/entity/Environment";
+import { EnvironmentRepository } from "../../repository/EnvironmentRepository";
+import { ApiGatewayEventParsed } from "../../utils/eventHandlingTool";
+import { APIGatewayProxyResult, Context } from "aws-lambda";
+import { ApiSuccessResponse, SuccessStatusCode } from "../../utils/response";
+import middy from "@middy/core";
+import xssSanitizer from "../../utils/xssSanitizer";
+import jsonBodyParser from "@middy/http-json-body-parser";
+import validator from "@middy/validator";
+import JSONErrorHandlerMiddleware from "middy-middleware-json-error-handler";
+import cors from "@middy/http-cors";
+import { CORS_CONFIGURATION } from "../../utils/corsConfig";
+import { errorHandlingMiddleware } from "../../utils/errorHandlingMiddleware";
+import { IpCheckerMiddleware } from "../../utils/ipLogging";
+import { validateRequestShape } from "../../utils/shapeValidator";
+import { uniqueNameValidator } from "../../utils/businessRulesValidation";
+import { environmentSchema } from "./createEnvironment";
+import { wrapSchema } from "../../utils/schemaWrapper";
+
+/**
+ * Submits an update to an environment of an application
+ *
+ * @param event - The PUT request from API Gateway
+ */
+export async function baseHandler(
+  event: ApiGatewayEventParsed<IEnvironment>,
+  context?: Context
+): Promise<APIGatewayProxyResult> {
+  const setupResult = validateRequestShape<IEnvironment>(event);
+  const { portfolioId, applicationId, environmentId } = setupResult.path;
+  const requestBody = setupResult.bodyObject;
+
+  // set up database connection
+  const connection = await createConnection();
+  let response: Environment;
+
+  try {
+    // ensures the portfolio, application, and environment id exists
+    await connection.getRepository(Portfolio).findOneOrFail({
+      id: portfolioId,
+    });
+    const application = await connection.getRepository(Application).findOneOrFail({
+      id: applicationId,
+    });
+    const environment = await connection
+      .getCustomRepository(EnvironmentRepository)
+      .findOneOrFail({ id: environmentId });
+
+    const environmentNames = await connection
+      .getCustomRepository(EnvironmentRepository)
+      .getAllEnvironmentNames(application.id);
```

Comment:
Since these are both reads before the update, we could probably run them in parallel with await `Promise.all()`.  https://stackoverflow.com/a/35612484

Reply:
I think this should definitely be considered to run these calls in parallel. I originally considering doing that with the calls to check for the `ids`, but the error handling was thrown off slightly so I removed them. To keep the errors clean for now I left this as-is, but as a part of refactoring we should try to run calls in parallel and figure out a nice way to deal with `error` handling when a `reject` does happen in `Promise.all()`.